File: 01a_[Individual] Data parsing servers - Part I/data_parsing.js
import { readFileSync } from "fs";
import { join } from "path";

const BASE_FOLDER = "../02_Text-based_Data_Formats";
const fileTypes = [".json", ".csv", ".xml", ".yaml", ".txt"];

fileTypes.forEach(ext => {
    const filePath = join(BASE_FOLDER, `me${ext}`);
    console.log(`\n--- me${ext} ---`);
    console.log(readFileSync(filePath, "utf-8"));
});

File: 01a_[Individual]_Data_parsing_servers_-_Part_I/data_parsing.py
from pathlib import Path

BASE_FOLDER = Path(__file__).parent.parent / "02_Text-based_Data_Formats"
file_types = [".json", ".csv", ".xml", ".yaml", ".txt"]

for ext in file_types:
    file_path = BASE_FOLDER / f"me{ext}"
    print(f"\n--- {file_path.name} ---")
    print(file_path.read_text(encoding="utf-8"))